### PR TITLE
phosh: 0.17 -> 0.21

### PR DESCRIPTION
**Pull request by tomfitzhenry**
###### Description of changes

Updates:
* Phosh to https://gitlab.gnome.org/World/Phosh/phosh/-/releases/v0.21.0
* Phoc to https://gitlab.gnome.org/World/Phosh/phoc/-/tags/v0.21.0

This PR also removes phoc's dependency on wlroots 0.14: https://github.com/NixOS/nixpkgs/issues/151311.

Tested by running on Pinephone Pro.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by zhaofengli** [comment]
A more complete set of patches can be found in https://github.com/zhaofengli/nixpkgs/commit/cf4cccece47442fa2c3c75b88a6d121a5af9f95f. I bumped both phoc and phosh in my local branch, and was intending to submit a PR after 0.20 was actually released. Could you add the other patches?

**Comment by zhaofengli** [comment]
I think it looks good. Will test on my PinePhone today and report back.

**Comment by zhaofengli** [comment]
Sorry for the delay. I just updated my PinePhone and now get a blank screen with this PR. I haven't dug into why it's happening though.

**Comment by hall** [comment]
I'm not entirely familiar with the testing process but just wanted to add that when I switch to `inputs.nixpkgs.url = github:tomfitzhenry/nixpkgs/phoc-0.20;` [my config](https://gitlab.com/hall/hall/-/blob/6749c426a2f3108a309e658b1f5f4f05532373b1/hosts/linux/aarch64/pinephone/configuration.nix) (for comparison) boots fine with the updated packages (thanks @tomfitzhenry :100: ).

Also, just saw a notification that [phosh v0.21.0](https://gitlab.gnome.org/World/Phosh/phosh/-/releases/v0.21.0) was released.

File: pkgs/applications/window-managers/phosh/default.nix [hunk 1]
```diff
@@ -85,19 +85,6 @@ stdenv.mkDerivation rec {
 
   mesonFlags = [ "-Dsystemd=true" "-Dcompositor=${phoc}/bin/phoc" ];
 
-  patches = [
-    # build: Adjust to polkit version changes
-    (fetchpatch {
```
Comment (SuperSandro2000) [inline pkgs/applications/window-managers/phosh/default.nix:90]: Please remove the unused fetchpatch from inputs.

File: pkgs/applications/misc/phoc/default.nix [hunk 2]
```diff
@@ -21,44 +21,42 @@
...
-      })
-      # xwayland: Allow to retrieve startup-id via _NET_STARTUP_INFO
+
+      # seat: Allow to cancel touches
       (fetchpatch {
-        name = "allow-to-retrieve-startup-id-via-net-startup-info.patch";
-        url = "https://github.com/swaywm/wlroots/commit/235bb6f2fcb8ee4174215ba74b5bc2f191c5960a.patch";
-        sha256 = "sha256-7AWBq12tF/781CmgvTaOvTIiiJMywxRn6eWp+jacdak=";
+        name = "seat-Allow-to-cancel-touches.patch";
+        url = "https://gitlab.freedesktop.org/wlroots/wlroots/-/commit/17b2b0663372.patch";
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/phoc/default.nix:42]: Please use a long commit sha.

File: pkgs/applications/misc/phoc/default.nix [hunk 3]
```diff
@@ -21,44 +21,42 @@
...
       })
 
-      # xwayland: Allow to retrieve _NET_STARTUP_ID
+      # xdg-activation: Deduplicate token creation code
       (fetchpatch {
-        name = "allow-to-retrieve-net-startup-id.patch";
-        url = "https://github.com/swaywm/wlroots/commit/66593071bc90a1cccaeedc636eb6f33c973f5362.patch";
-        sha256 = "sha256-yKf/twdUzrII5IakH7AH6LGyPDo9Nl/gIB0pTThSTfY=";
+        name = "xdg-activation-deduplicate-token-creation-code.patch";
+        url = "https://gitlab.freedesktop.org/wlroots/wlroots/-/commit/dd03d839ab56.patch";
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/phoc/default.nix:35]: same here

File: pkgs/applications/misc/phoc/default.nix [hunk 4]
```diff
@@ -21,44 +21,42 @@
...
       (fetchpatch {
-        name = "allow-to-retrieve-startup-id-via-net-startup-info.patch";
-        url = "https://github.com/swaywm/wlroots/commit/235bb6f2fcb8ee4174215ba74b5bc2f191c5960a.patch";
-        sha256 = "sha256-7AWBq12tF/781CmgvTaOvTIiiJMywxRn6eWp+jacdak=";
+        name = "seat-Allow-to-cancel-touches.patch";
+        url = "https://gitlab.freedesktop.org/wlroots/wlroots/-/commit/17b2b0663372.patch";
+        sha256 = "sha256-BAeXa3ZB5TXnlq0ZP2+rZlVXEPWpLP4Wi4TLwoXjkz4=";
       })
+
+      ./0001-handle-outputs-that-arent-in-the-layout.patch
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/phoc/default.nix:46]: Can we fetch this?

**Pull request merged**